src/VstsSyncMigrator.Core/Execution/MigrationContext/TestPlansAndSuitesMigrationContext.cs

```diff
@@ -622,27 +622,28 @@ private ITestSuiteBase FindSuiteEntry(IStaticTestSuite staticSuite, string title
             return testSuit;
         }
 
-        private ITestPlan FindTestPlan(TestManagementContext tmc, string name)
+        private ITestPlan FindTestPlan(string planName, int sourcePlanId)
         {
             Log.LogDebug("TestPlansAndSuitesMigrationContext::FindTestPlan");
-            ITestPlan testPlan = (from p in tmc.Project.TestPlans.Query("Select * From TestPlan") where p.Name == name select p).SingleOrDefault();
+            ITestPlan testPlan = (from p in _targetTestStore.Project.TestPlans.Query("Select * From TestPlan") where p.Name == planName select p).SingleOrDefault();
  
             if (testPlan != null)
             {
-                Log.LogDebug("TestPlansAndSuitesMigrationContext::FindTestPlan:: FOUND Test Plan with {name}", name);
+                Log.LogDebug("TestPlansAndSuitesMigrationContext::FindTestPlan:: FOUND Test Plan with {name}", planName);
                 //Check test plan is in fact the right one
-                var sourceWI = Engine.Source.WorkItems.GetWorkItem(testPlan.Id.ToString());
+                var sourceWI = Engine.Source.WorkItems.GetWorkItem(sourcePlanId);
                 string expectedReflectedId = Engine.Source.WorkItems.CreateReflectedWorkItemId(sourceWI).ToString();
-                string workItemReflectedId = (string)sourceWI.Fields[Engine.Target.Config.AsTeamProjectConfig().ReflectedWorkItemIDFieldName].Value;
+                var targetWI = Engine.Source.WorkItems.GetWorkItem(testPlan.Id.ToString());
```

Comment:
I believe this needs to use the `Target` Engine, like so

var targetWI = Engine.~~Source~~**Target**.WorkItems.GetWorkItem(testPlan.Id.ToString());

Reply:
Thats very strange as my code says as you write! Not sure what is going on...

![image](https://github.com/nkdAgility/azure-devops-migration-tools/assets/5205575/557cc410-13a7-4baf-b50b-c4ba5f3b39a5)